piccolo/query/methods/insert.py

```diff
@@ -70,16 +93,28 @@ def default_querystrings(self) -> t.Sequence[QueryString]:
 
         engine_type = self.engine_type
 
+        # TODO - check SQLite version
+        on_conflict = self.on_conflict_delegate._on_conflict
+        if on_conflict:
+            querystring = QueryString(
+                "{}{}",
+                querystring,
+                on_conflict.querystring,
+                query_type="insert",
+                table=self.table,
+            )
```

Comment:
```suggestion
        if engine_type in ("postgres", "cockroach") or (
            engine_type == "sqlite"
            and self.table._meta.db.get_version_sync() >= 3.24
        ):
            on_conflict = self.on_conflict_delegate._on_conflict
            if on_conflict:
                querystring = QueryString(
                    "{}{}",
                    querystring,
                    on_conflict.querystring,
                    query_type="insert",
                    table=self.table,
                )
        else:
            raise ValueError(
                "Sqlite versions lower than 3.24 are not supported"
            )
```

Reply:
Thanks. Having thought about it, I think we should move this check to the `Insert.on_conflict` method.

Also `NotImplementedError` is probably most appropriate as an exception.

```python
def on_conflict(...):
    if self.engine_type == 'sqlite' and self.table._meta.db.get_version_sync() < 3.34:
        raise NotImplementedError("SQLite versions lower than 3.24 are not supported") 
    ...
```

Reply:
I pushed this change, and checked for other similar methods which are only available in certain databases - they should all return `NotImplementedError`.

Reply:
Yes, that be much better.